common/src/main/java/tachyon/security/LoginUser.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.security.authentication.AuthenticationFactory;
+import tachyon.security.login.TachyonJaasConfiguration;
+
+/**
+ * A Singleton of LoginUser, which is an instance of {@link tachyon.security.User}. It represents
+ * the user of Tachyon client, when connecting to Tachyon service.
```

Comment:
Hi @calvinjia , LoginUser is not confined only to the client side. The master side will also use the
LoginUser to determine the root user for Tachyon filesystem. For example, when the tachyon creates the root inode, it should determines the user for the root inode.